.github/workflows/doc.yml

```diff
@@ -0,0 +1,58 @@
+name: doc
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  SEGMENT_DOWNLOAD_TIMEOUT_MINS: 1 # abort caching attempt if it's slow
+
+permissions:
+  contents: write
+
+jobs:
+  doc:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - run: sudo mkdir -p /nix
+    - run: sudo chown $(whoami) /nix
+    - uses: actions/cache@v3
+      with:
+        path: /nix
+        key: nix
+    - uses: nixbuild/nix-quick-install-action@v22
+      with:
+        nix_conf: |
+          experimental-features = nix-command flakes
+          access-tokens = github.com=${{ github.token }}
+          substituters = https://cache.nixos.org https://cache.garnix.io
+          trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g=
+    - uses: cachix/cachix-action@v12
+      continue-on-error: true
+      with:
+        name: wasmcloud
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+    - run: nix build -L .#checks.x86_64-linux.doc
+    - run: tar xf ./result/target.tar.zst
+    - name: Create `.nojekyll`
+      run: touch target/doc/.nojekyll
+    - name: Write `index.html`
+      run: |
+        cat <<EOF > target/doc/index.html
+          <!DOCTYPE html>
+          <meta charset="utf-8">
+          <title>Redirecting to wasmcloud/index.html</title>
+          <meta http-equiv="refresh" content="0; URL=wasmcloud/index.html">
+          <link rel="canonical" href="https://${{ github.repository.owner }}.github.io/wasmCloud/wasmcloud/index.html">
+        EOF
+    - uses: JamesIves/github-pages-deploy-action@v4.4.1
+      with:
+        folder: target/doc
+        clean: true
```

Comment:
If you're interested in not pushing to a branch, you can now deploy straight from a github action.

Upload pages as an artifact in the doc-build step. Then use deploy-pages
```yaml
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1

  # Deployment job
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
```

Reply:
Oh, cool, I'll try that!
Does that mean we also do not need `.nojekyll` thing anymore?

Reply:
Extra details here: https://github.com/actions/deploy-pages

I don't have a preference either way

Reply:
updated with the action, thanks for the suggestion!